src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -447,6 +447,18 @@ private void validateCvrFileSources() {
                 cvrPath);
           }
         }
+
+        if (isNullOrBlank(getContestId()) && isHart(source)) {
+          isValid = false;
+          Logger.log(
+              Level.SEVERE,
+              "contestId is required for Hart files.");
+        } else if (!isNullOrBlank(getContestId()) && !isHart(source)) {
```

Comment:
Is there really a reason to enforce this? If so, please update the .txt file to note that it's not valid with anything other than a Hart input.

Reply:
I think it's good to prevent the user from entering a field that can't possibly be relevant. I'll update the .txt.